src/components/typeahead/typeahead.component.ts

```diff
@@ -350,9 +354,7 @@ export class TypeaheadComponent<T = any> implements OnChanges, OnDestroy {
      * Display the first item as highlighted when there are several pages
      */
     onLoadedHighlight(): void {
-        // Clear previous highlight
-        this.highlighted$.next(null);
-        if (this.selectFirst && this.options) {
+        if (this.selectFirst && this.options && this._currentPageNumber === 0) {

```

Comment:
I think instead of having to add the `_currentPageNumber` variable, the `loaded` event which calls this function emits an object the contains the page that was loaded, we can us it instead.